core-addon/IonCore.js

```diff
@@ -166,4 +204,42 @@ module.exports = class IonCore {
         console.error(`IonCore._sendEnrollmentPing failed - error: ${error}`);
       });
   }
+
+  /**
+   * Sends a Pioneer enrollment ping.
+   *
+   * The `creationDate` provided by the telemetry APIs will be used as the
+   * timestamp for considering the user enrolled in pioneer and/or the study.
+   *
+   * @param {String} [studyAddonid=undefined]
+   *        optional study id. It's sent in the ping, if present, to signal
+   *        that user enroled in the study.
+   */
+  async _sendEnrollmentPing(studyAddonId) {
+    // If we were provided with a study id, then this is an enrollment to a study.
+    // Send the id alongside with the data and change the schema namespace to simplify
+    // the work on the ingestion pipeline.
+    if (typeof studyAddonId != "undefined") {
+      return await this._sendEmptyPing("pioneer-enrollment", studyAddonId);
+    }
+
+    // Note that the schema namespace directly informs how data is segregated after ingestion.
+    // If this is an enrollment ping for the pioneer program (in contrast to the enrollment to
+    // a specific study), use a meta namespace.
+    return await this._sendEmptyPing("pioneer-enrollment", "pioneer-meta");
+  }
+
+  /**
+   * Sends a Ion deletion-request ping.
+   *
+   * @param {String} studyAddonid
+   *        It's sent in the ping to signal that user unenroled from a study.
+   */
+  async _sendDeletionPing(studyAddonId) {
+    if (typeof studyAddonId == "undefined") {
```

Comment:
I think [strict equality operator for checking undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) is better here since you don't need to use a string (and have a potential typo etc). `(studyAddonId === undefined)` will throw if you misspell "undefined" in other words, while `typeof studyAddonId == "undefnied"` will just do the wrong thing :)

Reply:
OH! Got it! That's a great tip, thanks for the explanation!